geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -1298,6 +1301,26 @@ public boolean removeFromTempQueueEvents(Object tailKey) {
     }
   }
 
+  public boolean markAsDuplicateInTempQueueEvents(Object tailKey) {
+    synchronized (queuedEventsSync) {
+      Iterator<TmpQueueEvent> itr = tmpQueuedEvents.iterator();
+      while (itr.hasNext()) {
```

Comment:
For each loop?

Reply:
We are searching for first occurrence of this tailKey in loop.

Reply:
Yes, but I don't see where `itr` is used outside the loop where anything other than `next()` is invoked. You early exit the loop with returns. This can all be done in a for each loop.

Reply:
Alternatively this could be completed with streams.

Reply:
Iterator replaced with for loop. This function was based on function removeFromTempQueueEvents().